crates/miden-objects/src/note/note_tag.rs

```diff
@@ -108,13 +124,13 @@ impl NoteTag {
                 // [2 zero bits | remaining high bits (30 bits)].
                 let high_bits = high_bits as u32;
 
-                // Select the upper half of the u32 which then contains the 14 most significant bits
-                // of the account ID, i.e.:
-                // [2 zero bits | remaining high bits (14 bits) | 16 zero bits].
+                // // Select the upper half of the u32 which then contains the 14 most significant
+                // bits // of the account ID, i.e.:
+                // // [2 zero bits | remaining high bits (14 bits) | 16 zero bits].
                 let high_bits = high_bits & 0xffff0000;
 
                 // Set the local execution tag in the two most significant bits.
-                Ok(Self(high_bits | LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED))
+                Ok(Self::LocalAccount(high_bits | (LOCAL_ACCOUNT >> 30)))
```

Comment:
This set the least significant byte to `0xC0`, which is not what we want. If you remove the `>> 30` it should be correct.

`NoteTag::from` does the same thing for I think most variants. Can you double-check that?